shared/functions/componentDatalayer.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Function to track React component data and store it in a global object.
+ * @param {Object} args - An object containing the following named arguments:
+ *   @param {string} componentName - The name of the React component.
+ *   @param {string} package - The name of the package the component comes from.
+ *   @param {string} component - The initialized component's name.
+ *   @param {string} type - The component variation (e.g., "NA" if it does not apply).
+ *   @param {Object} configuration - Configuration data for the component.
+ */
+function trackReactComponent({
+  componentName = "",
+  package = "",
+  component = "",
+  type = "",
+  configuration = {}
+}) {
+  if (!componentName || !package || !component || !configuration) {
```

Comment:
Remove `configuration` check req another way

with the default `configuration = {}`, `!configuration` will always be `false` unless the user passes in a value like false (which wouldn't make sense)